### PR TITLE
[SPARK-31870][SQL][TESTS] Fix "Do not optimize skew join if additional shuffle" test having no skew join

### PR DESCRIPTION
**Pull request by manuzhang**
### What changes were proposed in this pull request?
Fix configurations and ensure there is skew join in the test "Do not optimize skew join if additional shuffle".

### Why are the changes needed?
The existing "Do not optimize skew join if additional shuffle" test has no skew join at all.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Fixed existing test.

### CODE REVIEW

**Comment by manuzhang** [comment]
cc @cloud-fan @JkSelf

**Comment by cloud-fan** [comment]
ok to test

File: sql/core/src/test/scala/org/apache/spark/sql/execution/adaptive/AdaptiveQueryExecSuite.scala [hunk 1]
```diff
@@ -639,11 +640,18 @@ class AdaptiveQueryExecSuite
           .range(0, 1000, 1, 10)
           .selectExpr("id % 1 as key2", "id as value2")
           .createOrReplaceTempView("skewData2")
-        val (_, innerAdaptivePlan) = runAdaptiveAndVerifyResult(
-          "SELECT key1 FROM skewData1 join skewData2 ON key1 = key2 group by key1")
+
+        def checkSkewJoin(query: String, additionalShuffle: Boolean): Unit = {
```
Comment (cloud-fan) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/adaptive/AdaptiveQueryExecSuite.scala:644]: nit: `optimizeSkewJoin: Boolean` to be more clear.

**Comment by maropu** [comment]
retest this please

File: sql/core/src/test/scala/org/apache/spark/sql/execution/adaptive/AdaptiveQueryExecSuite.scala [hunk 2]
```diff
@@ -639,11 +640,18 @@ class AdaptiveQueryExecSuite
...
+          assert(innerSmj.size == 1 && innerSmj.head.isSkewJoin != additionalShuffle)
+        }
+
+        checkSkewJoin(
+          "SELECT key1 FROM skewData1 join skewData2 ON key1 = key2", false)
         // Additional shuffle introduced, so disable the "OptimizeSkewedJoin" optimization
-        val innerSmj = findTopLevelSortMergeJoin(innerAdaptivePlan)
-        assert(innerSmj.size == 1 && !innerSmj.head.isSkewJoin)
+        checkSkewJoin(
+          "SELECT key1 FROM skewData1 join skewData2 ON key1 = key2 group by key1", true)
```
Comment (maropu) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/adaptive/AdaptiveQueryExecSuite.scala:654]: nit: Plz use uppercases for SQL keywords where possible (e.g., `group by` -> `GROUP BY`).

**Comment by manuzhang** [comment]
@cloud-fan 
Is joining tables of partitions [660, 720] and [720, 0]  with target size 100 a skew join ?

**Comment by cloud-fan** [comment]
it's not. the median size of the right table is `360`, and `720` is only 2 times of the median size.

**Comment by cloud-fan** [comment]
thanks, merging to master/3.0!

**Pull request closed without merging**